File: leetcode/1688 Count of Matches in Tournament/solution.go
func numberOfMatches(n int) int { // 方法一：模拟！！！
    teamNum, matches, result := n, 0, 0 // 数学方法请见Java版本答案！！！
    for(teamNum >= 2) {
        if teamNum%2 == 1 {
            matches = (teamNum-1) / 2
            teamNum = (teamNum-1) / 2 + 1
        } else {
            matches = teamNum / 2
            teamNum /= 2
        }
        result += matches
    }
    return result
}

File: leetcode/1688 Count of Matches in Tournament/solution.java
class Solution { // 方法二：数学！！！
    public int numberOfMatches(int n) { // 模拟方法请见Go语言版本答案！！！
        return n-1; // 按题意只有经过一场比赛，一支队伍才会被淘汰；由于最后只决出一个胜者，必然需要淘汰n-1支队伍，即必然进行n-1场比赛！！！
    }
}